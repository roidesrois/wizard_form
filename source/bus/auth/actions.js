// Types
import { types } from './types';

export const authActions = {
    // Sync
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
    valid: () => ({
        type: types.VALID,
    }),
    registered: () => ({
        type: types.REGISTERED,
    }),

    // Async
    initializeAsync: () => ({
        type: types.INITIALIZE_ASYNC,
    }),
    loginAsync: (credentials) => ({
        type:    types.LOGIN_ASYNC,
        payload: credentials,
    }),
    authenticateAsync: () => ({
        type: types.AUTHENTICATE_ASYNC,
    }),
    signupAsync: (userData) => ({
        type:    types.SIGNUP_ASYNC,
        payload: userData,
    }),
    confirmSignupAsync: (registrationHash) => ({
        type:    types.CONFIRM_SIGNUP_ASYNC,
        payload: registrationHash,
    }),
    activateAccountAsync: (userData) => ({
        type:    types.ACTIVATE_ACCOUNT_ASYNC,
        payload: userData,
    }),
    logoutAsync: () => ({
        type: types.LOGOUT_ASYNC,
    }),
};
