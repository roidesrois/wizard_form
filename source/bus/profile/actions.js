// Types
import { types } from './types';

export const profileActions = {
    // Sync
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
    updateAvatar: (newAvatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: newAvatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),

    // Async
    updateNameAsync: (newName) => ({
        type:    types.UPDATE_NAME_ASYNC,
        payload: newName,
    }),
    updateAvatarAsync: (newAvatar) => ({
        type:    types.UPDATE_AVATAR_ASYNC,
        payload: newAvatar,
    }),
    updatePasswordAsync: (newPassword) => ({
        type:    types.UPDATE_PASSWORD_ASYNC,
        payload: newPassword,
    }),
    updateEmailAsync: (newEmail) => ({
        type:    types.UPDATE_EMAIL_ASYNC,
        payload: newEmail,
    }),
    resetPasswordAsync: (email) => ({
        type:    types.RESET_PASSWORD_ASYNC,
        payload: email,
    }),
};
