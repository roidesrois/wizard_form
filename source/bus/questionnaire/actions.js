// Types
import { types } from './types';

export const questionnaireActions = {
    // Sync
    fillQuestionnaire: (questionnaire) => ({
        type:    types.FILL_QUESTIONNAIRE,
        payload: questionnaire,
    }),

    // setQuestionnaireValidation: (validation) => ({
    //     type:    types.SET_QUESTIONNAIRE,
    //     payload: validation,
    // }),

    // Async
    fetchQuestionnaireAsync: () => ({
        type: types.FETCH_QUESTIONNAIRE_ASYNC,
    }),
};
