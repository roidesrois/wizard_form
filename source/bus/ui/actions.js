// Core
import { v4 } from 'uuid';

// Instruments
import { types } from './types';

export const uiActions = {
    startFetching: () => ({
        type: types.START_FETCHING,
    }),
    stopFetching: () => ({
        type: types.STOP_FETCHING,
    }),
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    showNotification: (message, type = 'info', source = '') => ({
        type:    types.SHOW_NOTIFICATION,
        payload: {
            id: v4(),
            message,
            type,
            source,
        },
    }),
    hideNotification: () => ({
        type: types.HIDE_NOTIFICATION,
    }),
    setOnlineState: () => ({
        type: types.SET_ONLINE_STATE,
    }),
    setOfflineState: () => ({
        type: types.SET_OFFLINE_STATE,
    }),
};
