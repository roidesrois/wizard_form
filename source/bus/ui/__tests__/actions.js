// Core
import { uiActions } from '../actions';

const {
    testData: { error },
} = global;

describe('ui actions:', () => {
    test('startFetching snapshot', () => {
        expect(uiActions.startFetching()).toMatchSnapshot();
    });

    test('stopFetching snapshot', () => {
        expect(uiActions.stopFetching()).toMatchSnapshot();
    });

    test('emitError snapshot', () => {
        expect(uiActions.emitError(error, 'meta info')).toMatchSnapshot();
    });

    test('emitError with fallback params snapshot', () => {
        expect(uiActions.emitError(error, void 0)).toMatchSnapshot();
    });

    test('showNotification snapshot', () => {
        expect(
            uiActions.showNotification('login fail', 'error', 'login worker'),
        ).toMatchSnapshot();
    });

    test('showNotification with fallback params snapshot', () => {
        expect(
            uiActions.showNotification(void 0, 'login fail', void 0),
        ).toMatchSnapshot();
    });

    test('hideNotification', () => {
        expect(uiActions.hideNotification('123')).toMatchSnapshot();
    });

    test('setOnlineState snapshot', () => {
        expect(uiActions.setOnlineState()).toMatchSnapshot();
    });

    test('setOfflineState snapshot', () => {
        expect(uiActions.setOfflineState()).toMatchSnapshot();
    });
});
