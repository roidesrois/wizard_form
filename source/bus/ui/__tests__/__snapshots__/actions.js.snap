// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: emitError snapshot 1`] = `
Object {
  "error": true,
  "meta": "meta info",
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions: emitError with fallback params snapshot 1`] = `
Object {
  "error": true,
  "meta": null,
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions: hideNotification 1`] = `
Object {
  "type": "HIDE_NOTIFICATION",
}
`;

exports[`ui actions: setOfflineState snapshot 1`] = `
Object {
  "type": "SET_OFFLINE_STATE",
}
`;

exports[`ui actions: setOnlineState snapshot 1`] = `
Object {
  "type": "SET_ONLINE_STATE",
}
`;

exports[`ui actions: showNotification snapshot 1`] = `
Object {
  "payload": Object {
    "id": "123",
    "message": "login fail",
    "source": "login worker",
    "type": "error",
  },
  "type": "SHOW_NOTIFICATION",
}
`;

exports[`ui actions: showNotification with fallback params snapshot 1`] = `
Object {
  "payload": Object {
    "id": "123",
    "message": undefined,
    "source": "",
    "type": "login fail",
  },
  "type": "SHOW_NOTIFICATION",
}
`;

exports[`ui actions: startFetching snapshot 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`ui actions: stopFetching snapshot 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
